### PR TITLE
test: add cli_test for fswap module

### PR DESCRIPTION
**Pull request by 170210**
## Description

closes: #XXXX

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/fswap/client/testutil/tx.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+				s.toDenom.Base,
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1))).String()),
+			},
+			true,
+			0,
+			nil,
+		},
+		{
```
Comment (0Tech) [inline x/fswap/client/testutil/tx.go:86]: Is this case relevant to the coverage of x/fswap?

File: x/fswap/client/cli/tx.go [hunk 2]
```diff
@@ -22,6 +22,10 @@ const (
 	FlagSwapRate            = "swap-rate"
 )
 
+type ToDenomMeta struct {
+	Metadata bank.Metadata `json:"metadata"`
+}
+
```
Comment (jaeseung-bae) [inline x/fswap/client/cli/tx.go:28]: In general, It's not desirable to make it public for test.
Is there any way to test without exposing it?

Comment (170210) [reply x/fswap/client/cli/tx.go:28]: Do you have some ideas?

File: x/fswap/client/testutil/tx.go (hunk 1 above)
Comment (170210) [reply x/fswap/client/testutil/tx.go:86]: I want to simulate a situation that `coin amount` is invalid. Or I change it into `1abc1`?

File: x/fswap/client/cli/tx.go (hunk 2 above)
Comment (jaeseung-bae) [reply x/fswap/client/cli/tx.go:28]: Please wait, let me check.

Comment (jaeseung-bae) [reply x/fswap/client/cli/tx.go:28]: First you need to revert `ToDenomMeta` to private, then you can change `TestMsgSetSwap` code like below.

```go
func (s *IntegrationTestSuite) TestMsgSetSwap() {
	val := s.network.Validators[0]
	clientCtx := val.ClientCtx
	// avoid printing as yaml from CLI command
	clientCtx.OutputFormat = jsonOutputFormat

	denomMeta := struct {
		Metadata bank.Metadata `json:"metadata"`
	}{
		Metadata: s.toDenom,
	}
```

Comment (jaeseung-bae) [reply x/fswap/client/cli/tx.go:28]: Using the same tag(`json:"metadata"`) is enough to run this test.

Comment (170210) [reply x/fswap/client/cli/tx.go:28]: Yes, but this way defining the same structure in two places indeed increases the maintenance cost, as any changes to the structure need to be manually synchronized across both definitions. Is this acceptable?

Comment (jaeseung-bae) [reply x/fswap/client/cli/tx.go:28]: I can't come up another solution for this but exposing code for testing is a lot worse.

File: x/fswap/client/testutil/tx.go (hunk 1 above)
Comment (0Tech) [reply x/fswap/client/testutil/tx.go:86]: I mean, the case for "not enough fees" seems like redundant.

Comment (0Tech) [reply x/fswap/client/testutil/tx.go:86]: Also for "not enough gas".

File: x/fswap/client/cli/tx.go (hunk 2 above)
Comment (0Tech) [reply x/fswap/client/cli/tx.go:28]: I also have same opinion. Exposing implementation detail as public API is usually bad idea. Because, after exposing it, we cannot change them or the change must be treated as API breaking change.

File: x/fswap/client/testutil/tx.go (hunk 1 above)
Comment (170210) [reply x/fswap/client/testutil/tx.go:86]: These two tests are referenced by the bank module tests. I think they also can be deleted, I will modify it.

**Review by 0Tech — APPROVED** [review]

**Review by jaeseung-bae — APPROVED** [review]
